core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCache.java

```diff
@@ -156,6 +166,20 @@ public void process(WatchedEvent event) {
 
       switch (event.getType()) {
         case NodeDataChanged:
+          Matcher tableConfigsMatcher = TABLE_CONFIGS_PATTERN.matcher(event.getPath());
+          if (tableConfigsMatcher.matches()) {
+            try {
+              byte[] configToRefresh = getZooKeeper().getData(event.getPath(), watcher, null);
+              String strConfigToRefresh = new String(configToRefresh);
+              remove(strConfigToRefresh);
+              get(strConfigToRefresh);
```

Comment:
We are absolutely "resetting" the watcher here in the process method of ZCacheWatcher.  Watches are triggered once and then it removed from ZooKeeper.  Recreating and Resetting watches is synonymous in this context. 

The "table-config-version" znode doesn't have children so you won't see NodeChildrenChanged event on it ever.  It is created during initialization of the server and no watch can be put on it before that (in the ZooCache class at least)  so you won't see a NodeCreated event on the "table-config-version" znode.  We won't delete that node so you won't see a NodeDeleted event.   We sort of of rig it to only see events on it during NodeDataChanged because we change the data in it ourselves in the ZooUtil.  We do this the only time we care about and that is when a table config changes.